src/signal.rs

```diff
@@ -499,4 +505,89 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_get_blocked_signals() {
+        let mut sigset: sigset_t = unsafe { mem::zeroed() };
+
+        // Block a signal.
+        unsafe {
+            sigemptyset(&mut sigset);
+            sigaddset(&mut sigset, SIGRTMIN() + 0);
+            pthread_sigmask(SIG_BLOCK, &sigset, null_mut());
+        }
+
+        assert!(get_blocked_signals().unwrap().contains(&(SIGRTMIN() + 0)));
+    }
+
+    #[test]
+    fn test_block_signal() {
+        // Check if `SIGRTMIN() + 0` is blocked.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, SIGRTMIN() + 0), 0);
+        }
+
+        block_signal(SIGRTMIN() + 0).unwrap();
+
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, SIGRTMIN() + 0), 1);
+        }
+    }
+
+    #[test]
+    fn test_unblock_signal() {
```

Comment:
Can we merge this function with the previous one (we can maybe test unblocking the signal that we block there)?